src/nvim/optionstr.c

```diff
@@ -696,8 +696,11 @@ const char *did_set_buftype(optset_T *args)
       || opt_strings_flags(buf->b_p_bt, opt_bt_values, NULL, false) != OK) {
     return e_invarg;
   }
-  // buftype=prompt: set the prompt start position to lastline.
+  // buftype=prompt:
   if (buf->b_p_bt[0] == 'p') {
+    // Set default value for 'comments'
+    set_option_direct(kOptComments, STATIC_CSTR_AS_OPTVAL(""), OPT_LOCAL, SID_NONE);
```

Comment:
This is not a great place to do this but don't have a better answer atm.

And I don't think we can introduce `filetype=prompt`, because plugins often want to set e.g. `filetype=markdown` for prompt buffers.

Reply:
Technically filetypes are just running through auto-command. We can turn the buftype into psudo filetype with OptionSet event if we want to. Like add a prompt.vim/lua under ft plugin then load it with optionset event. In future we can add prompt related stuff there. Not sure if we want a psudo filetyoe file like that.

Reply:
no that would be worse :)

Reply:
it adds different problems, e.g. users need to know about `++nested`

and `OptionSet` is generally not something we want to depend on for core internals, especially 'buftype'. it's less established than `FileType` event. 

can revisit later if these cases continue to grow.

Reply:
Ok, I'm leaving it as for now.